### PR TITLE
Adding 'delete a post' feature

### LINKED ISSUE #117: Adding 'Delete post' feature

### PR DESCRIPTION
**Pull request by Shoval-Ovadia**
- created a class-based view for the feature - Post/views.py and Post/urls.py modified.
- created a template and for the HTML page - Post/templates/Post/deletePost.html created.
- Added an 'delete' button in 'post_detail.html' - Post/templates/Post/post_detail.html modified.

You can test this feature in the following way:

- Log in
- Create a new post
- Go to the list of posts
- Go to the details of the post you created (and you will see a 'delete' button).
- Delete your post and then check the post list.

closes #117

### CODE REVIEW

**Review by Yarboa — COMMENTED** [review]
Please add client test to test view/ template

**Comment by Shoval-Ovadia** [comment]
> Please add client test to test view/ template

Hi @Yarboa,
I added view, URL and general tests.

File: Post/views.py [hunk 1]
```diff
@@ -27,6 +28,18 @@ def CreateNewPost(request):
     return render(request, 'Post/createPost.html', {'form': form})
 
 
+class PostDeleteView(LoginRequiredMixin, UserPassesTestMixin, DeleteView):
+    model = Post
+    template_name = 'Post/deletePost.html'
+    success_url = '/postList/'
+
+    def test_func(self):
```
Comment (guy9050) [inline Post/views.py:36]: please rename this

File: Post/views.py [hunk 2]
```diff
@@ -27,6 +28,18 @@ def CreateNewPost(request):
...
+        post = self.get_object()
+        if self.request.user == post.user:
+            return True
+        return False
```
Comment (guy9050) [inline Post/views.py:40]: better to use `assert` instead of `if`

**Review by nadavspinco — APPROVED** [review]
The feature works well, after the changes that @guy9050 request will be made I think this should be merged.

File: Post/views.py (hunk 1 above)
Comment (Shoval-Ovadia) [reply Post/views.py:36]: This function needed to be implemented so 'UserPassesTestMixin' will run it.

File: Post/views.py (hunk 2 above)
Comment (Shoval-Ovadia) [reply Post/views.py:40]: @guy9050 I tried to use assert instead, but it didn't work that way

**Review by maayanhd — CHANGES_REQUESTED** [review]

**Comment by Shoval-Ovadia** [comment]
@maayanhd 
what do you mean?
this is a function that needs to be implemented and it needs to return true or false- it is the test that the class runs to make sure that the user that tries to delete the post - is the one who wrote it.

**Comment by maayanhd** [comment]
@Shoval-Ovadia You commented on @guy9050 question that you can't use the assert so I was asking why? what was the problem? on paper assert suppose to do the same thing, I'm trying to understand what was the error or the reason behind that.

File: Post/templates/Post/deletePost.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	<legend class="border-bottom mb-4">Delete post:</legend>
+	<form method="POST">
+		{% csrf_token %}
+		<h2>Are you sure you want to delete the post?</h2>
+		<button class="btn btn-outline-danger" type="submit">Yes, delete.</button>
+		<a class="btn btn-outline-secondary" href="{% url 'post_detail' object.id %}">Cancel</a>
+	</form>
+</div>
+
+{% endblock content %}
```
Comment (Leeadar) [inline Post/templates/Post/deletePost.html:14]: Please add a new line here

**Review by Leeadar — APPROVED** [review]

**Comment by Shoval-Ovadia** [comment]
> @Shoval-Ovadia You commented on @guy9050 question that you can't use the assert so I was asking why? what was the problem? on paper assert suppose to do the same thing, I'm trying to understand what was the error or the reason behind that.
> I know assert returns assertionerror and not false, that was the problem? And why didn't you just returned then self.request.user == post.user instead of the if and else structure?

It is don't do the work - I get '403 Forbidden' error.. 
this is the change I did:
image

File: Post/templates/Post/deletePost.html (hunk 3 above)
Comment (Shoval-Ovadia) [reply Post/templates/Post/deletePost.html]: Done!

File: Post/tests.py [hunk 4]
```diff
@@ -93,3 +95,58 @@ def form():
             'Description': description,
         }
     )
+
+
+class PostDeleteTest(TestCase):
```
Comment (Yarboa) [inline Post/tests.py:100]: Please use django-pytest convention review other PRs ot class talk related UI testing

File: Post/views.py (hunk 2 above)
Comment (guy9050) [reply Post/views.py:40]: @Shoval-Ovadia What was the problem

**Comment by guy9050** [comment]
Black test failing.

File: Post/tests.py (hunk 4 above)
Comment (Shoval-Ovadia) [reply Post/tests.py]: Done!

**Review by maayanhd — APPROVED** [review]

**Comment by Shoval-Ovadia** [comment]
> Black test failing.

Fixed!

**Review by guy9050 — APPROVED** [review]

**Pull request merged**